src/Command/DrupalUpdate/FileSystemUtility.php

```diff
@@ -0,0 +1,169 @@
+<?php
+namespace Acquia\Cli\Command\DrupalUpdate;
+
+use Acquia\Cli\Exception\AcquiaCliException;
+use PharData;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Class FileSystemUtility
+ * @package Acquia\Cli\Command\DrupalUpdate
+ */
+class FileSystemUtility
+{
+  /**
+   * @var
+   */
+  private Filesystem $fileSystem;
+
+  /**
+   * @var
+   */
+  private SymfonyStyle $io;
+
+  /**
+   * FileSystemUtility constructor.
+   * @param InputInterface $input
+   * @param OutputInterface $output
+   */
+  public function __construct(InputInterface $input,
+                                OutputInterface $output) {
+    $this->io = new SymfonyStyle($input, $output);
+    $this->fileSystem =  new Filesystem();
+  }
+
+  /**
+   * Download and extract tar files in given directory path.
+   * @param $package
+   * @param $file_url
+   * @param $save_to
+   */
+  function downloadRemoteFile($package, $file_url, $save_to) {
+    echo $package . "==" . $file_url . "==" . $save_to . PHP_EOL;
+    if ($package == 'drupal') {
+
+      $this->downloadRemoteFileDrupalCore($package, $file_url, $save_to);
+      return;
+    }
+
+    try {
+      $content = file_get_contents($file_url);
```

Comment:
Use guzzle or curl instead of file_get_contents

Reply:
Code updated using guzzleHttpClient.